301-spinnaker-vmss-or-aks/azuredeploy.json

```diff
@@ -0,0 +1,402 @@
+    "vmStorageAccountContainerName": "vhds",
+    "keyVaultName": "[concat('vault', uniqueString(resourceGroup().id))]",
+    "vnetName": "[concat(variables('scenarioPrefix'), 'Vnet')]",
+    "subnet0Name": "[concat(variables('scenarioPrefix'), 'Subnet0')]",
+    "subnet1Name": "[concat(variables('scenarioPrefix'), 'Subnet1')]",
+    "subnet2Name": "[concat(variables('scenarioPrefix'), 'Subnet2')]",
+    "frontEndNSGName": "[concat(variables('scenarioPrefix'), 'NSG')]",
+    "nicName": "[concat(variables('scenarioPrefix'), 'Nic')]",
+    "publicIPAddressName": "[concat(variables('scenarioPrefix'), 'PublicIP')]",
+    "vmName": "[concat(variables('scenarioPrefix'), 'VM')]",
+    "vmExtensionName": "[concat(variables('scenarioPrefix'), 'Init')]",
+    "_artifactsLocation": "https://raw.githubusercontent.com/Azure/azure-devops-utils/master/",
+    "_extensionScript": "101-spinnaker-install-selection.sh",
+    "_artifactsLocationSasToken": "",
+    "linuxConfiguration": {
+      "disablePasswordAuthentication": true,
+      "ssh": {
+        "publicKeys": [
+          {
+            "path": "[concat('/home/', parameters('adminUsername'), '/.ssh/authorized_keys')]",
+            "keyData": "[parameters('sshPublicKey')]"
+          }
+        ]
+      }
+    },
+    "adminVMPassword": "[if(equals(parameters('authenticationType'), 'password'), parameters('adminPassword'), variables('linuxConfiguration'))]",
+    "adminJenkinsPassword": "[if(empty(parameters('aksClusterName')), parameters('adminPassword'), '')]",
+    "useSshPublicKey": "[if(equals(parameters('authenticationType'), 'password'), 'false', 'true')]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[variables('storageAccountName')]",
+      "apiVersion": "2019-04-01",
+      "location": "[parameters('location')]",
+      "kind": "Storage",
+      "sku": {
+        "name": "Standard_LRS"
+      },
+      "properties": {}
+    },
+    {
+      "type": "Microsoft.KeyVault/vaults",
+      "apiVersion": "2018-02-14",
+      "name": "[variables('keyVaultName')]",
+      "location": "[parameters('location')]",
+      "resources": [
+        {
+          "type": "secrets",
+          "name": "VMUsername",
+          "apiVersion": "2018-02-14",
+          "dependsOn": [
+            "[resourceId('Microsoft.KeyVault/vaults', variables('keyVaultName'))]"
+          ],
+          "properties": {
+            "value": "[parameters('adminUsername')]"
+          }
+        },
+        {
+          "type": "secrets",
+          "name": "VMPassword",
+          "apiVersion": "2018-02-14",
+          "dependsOn": [
+            "[resourceId('Microsoft.KeyVault/vaults', variables('keyVaultName'))]"
+          ],
+          "properties": {
+            "value": "[if(equals(parameters('authenticationType'), 'password'), variables('adminVMPassword'), variables('adminVMPassword').ssh.publicKeys[0].keyData)]"
```

Comment:
can simplify this once the other change is done

Reply:
> can simplify this once the other change is done

We need to support our business scenario so that there is no need to change here.